unstructured/ingest/main.py

```diff
@@ -35,6 +36,35 @@
     mp.set_start_method("spawn")
 
 
+class MutuallyExclusiveOption(Option):
+    def __init__(self, *args, **kwargs):
+        self.mutually_exclusive = set(kwargs.pop("mutually_exclusive", []))
+        help = kwargs.get("help", "")
+        if self.mutually_exclusive:
+            mutex_str = ", ".join(self.mutually_exclusive)
+            kwargs["help"] = help + (
+                " NOTE: This argument is mutually exclusive with "
+                " arguments: [" + mutex_str + "]."
+            )
+        super(MutuallyExclusiveOption, self).__init__(*args, **kwargs)
+
+    def handle_parse_result(self, ctx, opts, args):
+        if self.mutually_exclusive.intersection(opts) and self.name in opts:
+            raise UsageError(
+                "Illegal usage: `{}` is mutually exclusive with "
+                "arguments `{}`.".format(
+                    self.name,
+                    ", ".join(self.mutually_exclusive),
+                ),
+            )
```

Comment:
I just tried:
```
python3 -m unstructured.ingest.main --gitlab-url https://gitlab.com/gitlab-com/content-sites/docsy-gitlab --git-file-glob '*.md,*.txt' --reprocess --metadata-exclude page_number --metadata-include page_number
```

This succeeded. I thought I should have seen this error message, right?